value/src/main/java/com/google/auto/value/AutoValueExtension.java

```diff
@@ -0,0 +1,101 @@
+package com.google.auto.value;
+
+import javax.annotation.processing.ProcessingEnvironment;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.TypeElement;
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * An AutoValueExtension allows for extra functionality to be created during the generation
+ * of an AutoValue class.
```

Comment:
Can you add a sentence:
NOTE: the design of this interface is not final and subject to change.